src/adapter/z-stack/znp/znp.ts

```diff
@@ -69,26 +69,22 @@ class Znp extends events.EventEmitter {
         this.unpiParser = new UnpiParser();
     }
 
-    private log(type: Type, message: string): void {
-        if (type === Type.SRSP) {
-            logger.debug(`SRSP: ${message}`, NS);
-        } else if (type === Type.AREQ) {
-            logger.debug(`AREQ: ${message}`, NS);
-        } else {
-            /* istanbul ignore else */
-            if (type === Type.SREQ) {
-                logger.debug(`SREQ: ${message}`, NS);
-            } else {
-                throw new Error(`Unknown type '${type}'`);
-            }
+    private static stringifyObject(prefix: string, object: ZpiObject): string {
+        return `${Type[object.type]}: ${prefix} ${Subsystem[object.subsystem]} - ${object.command} - ${JSON.stringify(object.payload)}`;
+    }
+
+    private log(prefix: string, object: ZpiObject): void {
```

Comment:
Should remove `stringifyObject` and `log` functions and declare `toString` in `ZpiObject`

```typescript
    public toString(direction: '<--' | '-->'): string {
        return `${Type[this.type]}: ${direction} ${Subsystem[this.subsystem]} - ${this.command} - ${JSON.stringify(this.payload)}`;
    }
```
then just call `logger.debug(() => object.toString('<--'), NS);` directly in the 2 places where it's used.

Reply:
I made the change, but since I don't think direction belongs to the object, I kept it outside of the toString().